core/src/main/scala/spinal/core/Area.scala

```diff
@@ -126,6 +126,17 @@ object ImplicitArea{
 class ClockingArea(val clockDomain: ClockDomain) extends Area with PostInitCallback {
   val ctx = ClockDomainStack.set(clockDomain)
 
+  /*
```

Comment:
Agreed.  I'm thinking that maybe it is possible to automatically derive the actual clockdomain (instead of relying on `Component.clockDomain`, which is in a lot of cases not the correct domain) from connection relationships?  Given that we can specify clock domain for input and output ports, we should be able to propagate this information along the path...

Reply:
I've had a look and every BaseType (which is every HW data type) has a member that is initialized with the CD on creation: https://github.com/SpinalHDL/SpinalHDL/blob/dev/core/src/main/scala/spinal/core/BaseType.scala#L62 though you should only use them if isReg is true

So you only need to search back to the previous register.